### PR TITLE
Justpremium Adapter for 1.0

### PR DESCRIPTION
**Pull request by mkalafior**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

- test parameters for validating bids
```
       {
           sizes: [[1, 1]],
           code: 'div-gpt-ad-1471513102552-0',
           bids: [
               {
                   bidder: 'justpremium', 
                   params: {
                       zone: 34364
                   }
               },
           ]
       }
```

- maintainer: sebastian.widelak@justpremium.com
- [x] official adapter submission

@matthewlane @ndhimehta @snapwich I created new pr as after merging latest master to branch used in previous one some files were just deleted causing that tests failed. This one contains exactly the same code, so we will be very grateful for quick approval.

### CODE REVIEW

File: modules/justpremiumBidAdapter.js [hunk 1]
```diff
@@ -1,274 +1,186 @@
...
-
-  function isCrossOriginIframe() {
-    try {
-      return !top.document;
-    } catch (e) {
-      return true;
+import { registerBidder } from 'src/adapters/bidderFactory'
+
+const BIDDER_CODE = 'justpremium'
+const ENDPOINT_URL = top.document.location.protocol + '//pre.ads.justpremium.com/v/2.0/t/xhr'
```
Comment (matthewlane) [inline modules/justpremiumBidAdapter.js]: `top` should generally be in a try/catch block, or use `utils.getTopWindowLocation()` to safely get `location` here

File: modules/justpremiumBidAdapter.js [hunk 2]
```diff
@@ -1,274 +1,186 @@
...
+  },
+
+  buildRequests: (validBidRequests) => {
+    const c = preparePubCond(validBidRequests)
+    const payload = {
+      zone: [...new Set(validBidRequests.map(b => {
+        return parseInt(b.params.zone)
+      }))].join(','),
+      hostname: top.document.location.hostname,
+      protocol: top.document.location.protocol.replace(':', ''),
```
Comment (matthewlane) [inline modules/justpremiumBidAdapter.js]: Can also use `utils.getTopWindowLocation()` in the above two lines

File: modules/justpremiumBidAdapter.js [hunk 3]
```diff
@@ -1,274 +1,186 @@
...
+      sw: window.top.screen.width,
+      sh: window.top.screen.height,
+      ww: window.top.innerWidth,
+      wh: window.top.innerHeight,
```
Comment (matthewlane) [inline modules/justpremiumBidAdapter.js]: The above four `window.top` references will need to be in a try/catch block, or can set up a variable that access `window.top`, place that within a try/catch, and access it here

File: test/spec/modules/justpremiumBidAdapter_spec.js [hunk 4]
```diff
@@ -1,386 +1,126 @@
...
-function createCookie(name, value, days) {
-  var expires = '';
-  if (days) {
-    var date = new Date();
-    date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
-    expires = '; expires=' + date.toUTCString();
-  }
-  document.cookie = name + '=' + value + expires + '; path=/';
-}
+import { expect, assert } from 'chai'
```
Comment (matthewlane) [inline test/spec/modules/justpremiumBidAdapter_spec.js]: `assert` isn't used, can be removed from import

File: test/spec/modules/justpremiumBidAdapter_spec.js [hunk 5]
```diff
@@ -1,386 +1,126 @@
...
+import { spec } from 'modules/justpremiumBidAdapter'
+import { getTopWindowLocation } from 'src/utils'
+import { newBidder } from 'src/adapters/bidderFactory'
```
Comment (matthewlane) [inline test/spec/modules/justpremiumBidAdapter_spec.js]: The above two imports aren't used in the test, can be removed

File: modules/justpremiumBidAdapter.js [hunk 6]
```diff
@@ -1,274 +1,186 @@
...
+
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL + '?i=' + (+new Date()),
+      data: payloadString,
+      bids: validBidRequests
+    }
+  },
+
+  interpretResponse: (serverResponse, bidRequests) => {
```
Comment (matthewlane) [inline modules/justpremiumBidAdapter.js]: #1748 changed the first argument of `interpretResponse` to:

```JavaScript
{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
```

so adding something like
```JavaScript
serverResponse = serverResponse.body;
```
just below this line, or however you'd prefer to grab the `body`, and updating corresponding tests if needed should get this back to working properly

File: modules/justpremiumBidAdapter.js [hunk 7]
```diff
@@ -1,274 +1,186 @@
...
+    let bidResponses = []
+    bidRequests.bids.forEach(adUnit => {
+      let bidId = adUnit.bidId
+      let bid = findBid(adUnit.params, serverResponse.bid)
+      if (bid) {
+        let size = (adUnit.sizes && adUnit.sizes.length && adUnit.sizes[0]) || []
+        let bidResponse = {
+          requestId: bidId,
+          bidderCode: spec.code,
```
Comment (matthewlane) [inline modules/justpremiumBidAdapter.js]: `bidderCode` will be set by `bidderFactory` automatically now, this line can be dropped

File: modules/justpremiumBidAdapter.js [hunk 8]
```diff
@@ -1,274 +1,186 @@
...
+          width: size[0] || bid.width,
+          height: size[1] || bid.height,
+          ad: bid.adm,
+          cpm: bid.price,
+          currency: bid.currency || 'USD',
+          ttl: bid.ttl || spec.time
```
Comment (matthewlane) [inline modules/justpremiumBidAdapter.js]: `netRevenue` and `creativeId` will also need to be in this object to pass validation, see the table in http://prebid.org/dev-docs/bidder-adapter-1.html#interpreting-the-response for a description of these fields

**Comment by mkalafior** [comment]
@matthewlane all requested changes were applied.

**Pull request merged**